modules/navier_stokes/doc/content/source/kernels/INSADMomentumGradDiv.md

```diff
@@ -0,0 +1,55 @@
+# INSADMomentumGradDiv
+
+This kernel implements grad-div stabilization for the purpose of building
+scalable preconditioners at high Reynolds numbers for Navier-Stokes as introduced in
+[!citep](benzi2006augmented). Its weak form is given by
+
+\begin{equation}
+\gamma\left(\textrm{div}\ \mathbf{u}_h, \textrm{div}\ \mathbf{v}_h\right)
+\end{equation}
+
+where $\gamma$ is a penalty parameter, $\mathbf{u}_h$ is the velocity vector
+approximate solution, and $\mathbf{v}_h$ are its associated test functions. For
+$\gamma$ not too small, the Schur complement inverse of a linearized (via Newton or
+Picard) stable (no pressure on-diagonal) Navier-Stokes discretization is well
+approximated by
+
+\begin{equation}
+S^{-1} = -\left(\nu + \gamma\right)M_p^{-1}
+\end{equation}
+
+where $\nu$ is the kinematic viscosity and $M_p$ is the pressure mass matrix
+([!citep](farrell2019augmented)). In [!citep](benzi2006augmented) the authors
+found that $\gamma$ on the order of the velocity magnitude provides $\nu$- and
+$h$-independent (the latter denoting element size) convergence for a wide range
```

Comment:
so gamma is ok as a constant? or do they make it a field

Reply:
it should be a constant